pkg/resources/nodeagent/nodeagent.go

```diff
@@ -0,0 +1,395 @@
+		if userDefined.Type != "fluentbit_windows" {
+			programDefault.FluentbitSpec.Metrics = &v1beta1.Metrics{
+				Port: 2020,
+				Path: "/",
+			}
+		}
+	}
+
+	if userDefined.FluentbitSpec.Metrics != nil {
+
+		programDefault.FluentbitSpec.Metrics = &v1beta1.Metrics{
+			Interval: "15s",
+			Timeout:  "5s",
+			Port:     2020,
+			Path:     "/api/v1/metrics/prometheus",
+		}
+		err := merge.Merge(programDefault.FluentbitSpec.Metrics, userDefined.FluentbitSpec.Metrics)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if programDefault.FluentbitSpec.Metrics != nil && userDefined.FluentbitSpec.Metrics != nil && userDefined.FluentbitSpec.Metrics.PrometheusAnnotations {
+		defaultPrometheusAnnotations := &typeoverride.ObjectMeta{
+			Annotations: map[string]string{
+				"prometheus.io/scrape": "true",
+				"prometheus.io/path":   programDefault.FluentbitSpec.Metrics.Path,
+				"prometheus.io/port":   fmt.Sprintf("%d", programDefault.FluentbitSpec.Metrics.Port),
+			},
+		}
+		err := merge.Merge(&(programDefault.FluentbitSpec.DaemonSetOverrides.Spec.Template.ObjectMeta), defaultPrometheusAnnotations)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if programDefault.FluentbitSpec.Metrics != nil {
+		defaultLivenessProbe := &v1.Probe{
+			Handler: v1.Handler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: programDefault.FluentbitSpec.Metrics.Path,
+					Port: intstr.IntOrString{
+						IntVal: programDefault.FluentbitSpec.Metrics.Port,
+					},
+				}},
+			InitialDelaySeconds: 10,
+			TimeoutSeconds:      0,
+			PeriodSeconds:       10,
+			SuccessThreshold:    0,
+			FailureThreshold:    3,
+		}
+		if programDefault.FluentbitSpec.DaemonSetOverrides.Spec.Template.Spec.Containers[0].LivenessProbe == nil {
+			programDefault.FluentbitSpec.DaemonSetOverrides.Spec.Template.Spec.Containers[0].LivenessProbe = &v1.Probe{}
+		}
+
+		err := merge.Merge(programDefault.FluentbitSpec.DaemonSetOverrides.Spec.Template.Spec.Containers[0].LivenessProbe, defaultLivenessProbe)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return programDefault, nil
+}
+
+var NodeAgentFluentbitWindowsDefaults = &v1beta1.NodeAgent{
+	FluentbitSpec: &v1beta1.NodeAgentFluentbit{
+		FilterKubernetes: v1beta1.FilterKubernetes{
+			KubeURL:       "https://kubernetes.default.svc.cluster.local:443",
+			KubeCAFile:    "c:\\var\\run\\secrets\\kubernetes.io\\serviceaccount\\ca.crt",
+			KubeTokenFile: "c:\\var\\run\\secrets\\kubernetes.io\\serviceaccount\\token",
+			KubeTagPrefix: "kubernetes.C.var.log.containers.",
+		},
+		InputTail: v1beta1.InputTail{
+			Path: "C:\\var\\log\\containers\\*.log",
+		},
+		ContainersPath: "C:\\ProgramData\\docker",
+		VarLogsPath:    "C:\\var\\log",
+		DaemonSetOverrides: &typeoverride.DaemonSet{
+			Spec: typeoverride.DaemonSetSpec{
+				Template: typeoverride.PodTemplateSpec{
+					Spec: typeoverride.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:    containerName,
```

Comment:
I have two questions here.
1. Is this going to be exposed via `values.yaml`?
2. Is the Dockerfile for this going to be included in this PR?
CC: @cbron

Reply:
1. yes
2. yes, we will include it as an example, just won't setup any automated build for it. once you have a build and image in place we can use and add it as the default windows image

Reply:
Okay great. I'll get started on that: https://github.com/rancher/fluent-bit-package